src/javascript/app_2/Stores/Modules/Contract/Helpers/__tests__/logic.js

```diff
@@ -0,0 +1,362 @@
+                chart_type: 'candle',
+                end_epoch: 1544025900,
+                start_epoch: 1543999100,
+            });
+        });
+        it('should work as expected with values leading to granularity 14400', () => {
+            const contract_info = {
+                "date_expiry":1546000000,
+                "date_start":1544000000,
+            };
+            expect(Logic.getChartConfig(contract_info)).to.eql({
+                granularity: 14400,
+                chart_type: 'candle',
+                end_epoch: 1546014400,
+                start_epoch: 1543985600,
+            });
+        });
+        it('should work as expected when duration is more than 30 * 24 * 3600', () => {
+            const contract_info = {
+                "date_expiry":1546592100,
+                "date_start":1544000000,
+            };
+            expect(Logic.getChartConfig(contract_info)).to.eql({
+                granularity: 86400,
+                chart_type: 'candle',
+                end_epoch: 1546678500,
+                start_epoch: 1543913600,
+            });
+        });
+    });
+
+    describe('isEnded', () => {
+        it('should return false when there is status and it\'s equal to open in contract info', () => {
+            const contract_info = {
+                "status": "open",
+            };
+            expect(Logic.isEnded(contract_info)).to.eql(false);
+        });
+        it('should return true when there is status and it\'s not equal to open in contract info', () => {
+            const contract_info = {
+                "status": "closed",
+            };
+            expect(Logic.isEnded(contract_info)).to.eql(true);
+        });
+        it('should return true when contract is expired', () => {
+            const contract_info = {
+                "status": "open",
+                "is_expired": true,
+            };
+            expect(Logic.isEnded(contract_info)).to.eql(true);
+        });
+        it('should return true when contract is settleable', () => {
+            const contract_info = {
+                "status": "open",
+                "is_expired": false,
+                "is_settleable": true,
+            };
+            expect(Logic.isEnded(contract_info)).to.eql(true);
+        });
+        it('should return true when contract is not expired', () => {
+            const contract_info = {
+                "status": "open",
+                "is_expired": false,
+            };
+            expect(Logic.isEnded(contract_info)).to.eql(false);
+        });
+        it('should return true when contract does not have is_settleable, is_expired and status', () => {
+            const contract_info = {};
+            expect(Logic.isEnded(contract_info)).to.eql(false);
+        });
+    });
+
+    expect('getDisplayStatus', () => {
+        it('should return won if contract is ended and profit is more than zero', () => {
+            const contract_info = {
+                "status": "closed",
```

Comment:
if not mistaken, valid contract statuses from the API are `open` `sold` `won` `lost`. maybe use `won` or `sold` here?